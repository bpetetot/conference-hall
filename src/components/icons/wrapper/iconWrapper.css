.cc-icon-small {
  width: 16px;
  height: 16px;
}

.cc-icon-medium {
  width: 24px;
  height: 24px;
}

.cc-icon-large {
  width: 32px;
  height: 32px;
}