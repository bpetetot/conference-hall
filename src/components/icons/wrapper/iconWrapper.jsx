import React from 'react'
import PropTypes from 'prop-types'
import cn from 'classnames'

import './iconWrapper.css'

const iconWrapper = (Icon) => {
  const Wrapper = ({
    className, medium, large, ...rest
  }) => {
    const classes = cn(
      'cc-icon cc-icon-small',
      {
        'cc-icon-medium': medium,
        'cc-icon-large': large,
      },
      className,
    )

    return <Icon className={classes} {...rest} />
  }

  Wrapper.propTypes = {
    className: PropTypes.string,
    medium: PropTypes.bool,
    large: PropTypes.bool,
  }

  Wrapper.defaultProps = {
    className: undefined,
    medium: undefined,
    large: undefined,
  }

  return Wrapper
}

export default iconWrapper
