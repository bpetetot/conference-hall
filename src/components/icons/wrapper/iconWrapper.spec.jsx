/* eslint-env jest */
import React from 'react'
import iconWrapper from './index'

describe('components/iconWrapper', () => {
  it('should render', () => {
    expect(<iconWrapper />).toMatchSnapshot()
  })
})
