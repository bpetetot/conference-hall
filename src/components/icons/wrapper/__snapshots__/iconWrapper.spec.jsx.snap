// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/iconWrapper should render 1`] = `<iconWrapper />`;
